infra/stream/Formatter.cpp

```diff
@@ -0,0 +1,225 @@
+#include "infra/stream/Formatter.hpp"
+#include "infra/stream/StreamErrorPolicy.hpp"
+#include <cctype>
+
+namespace infra
+{
+    FormatSpec::FormatSpec(const char*& format)
```

Comment:
Should we add some of these implementation helpers in a detail namespace? They are not meant to be used by consumers of this class, right?